src/scitokens/scitokens.py

```diff
@@ -74,13 +89,42 @@ def verify(self):
         raise NotImplementedError()
 
 
-    def serialize(self, include_key=False):
+    def serialize(self, include_key=False, issuer = None, lifetime = 600):
         """
         Serialize the existing SciToken.
+        
+        :return str: base64 encoded token
         """
         
+        if self._key == None:
+            raise MissingKeyException("Unable to serialize, missing private key")
+        
+        # Issuer needs to be available, otherwise throw an error
+        if issuer == None and 'iss' not in self._claims:
+            raise MissingIssuerException("Issuer not specific in claims or as argument")
+        
+        if not issuer:
+            issuer = self._claims['iss']
+        
+        issue_time = int(mktime(datetime.utcnow().timetuple()))
+        exp_time = int(mktime((datetime.utcnow() + timedelta(seconds=lifetime)).timetuple()))
+        
+        payload = dict(self._claims)
+        
+        # Anything below will override what is in the claims
+        payload.update({
+            "iss": issuer,
+            "exp": exp_time,
+            "iat": issue_time,
+        })
+        
+        encoded = jwt.encode(payload, self._key, algorithm = "RS256")
```

Comment:
Once you've signed the token, save it to a `_serialized_token` member.  Further, copy over all the claims to the verified claims dict.

Reply:
Done!